metadata-ingestion/src/datahub/ingestion/source/ldap.py

```diff
@@ -19,25 +25,33 @@ def create_controls(pagesize: int) -> SimplePagedResultsControl:
     return SimplePagedResultsControl(True, size=pagesize, cookie="")
 
 
-def get_pctrls(serverctrls):
+def get_pctrls(
+    serverctrls: List[SimplePagedResultsControl],
+) -> List[SimplePagedResultsControl]:
     """
     Lookup an LDAP paged control object from the returned controls.
     """
     return [
-        c for c in serverctrls if c.controlType == SimplePagedResultsControl.controlType
+        c
+        for c in serverctrls
+        if c.controlType == SimplePagedResultsControl.controlType
     ]
 
 
-def set_cookie(lc_object, pctrls, pagesize):
+def set_cookie(
+    lc_object: SimplePagedResultsControl,
+    pctrls: List[SimplePagedResultsControl],
+) -> bool:
     """
     Push latest cookie back into the page control.
     """
+
     cookie = pctrls[0].cookie
     lc_object.cookie = cookie
-    return cookie
+    return bool(cookie)
```

Comment:
So this type changed because mypy complained. cookie used to be a binary, but in get_workunits it's declared as a bool

Reply:
looks good - and thank you for adding all these type annotations 🙂